### DISCUSSION TITLE
Restore Android 4 support

### DISCUSSION THREAD

**Discussion by link2xt** [maplibre/maplibre-native#1011] [General]
API level was bumped from level 14 to 21 in #630, but in fact it breaks compatibility with all Android 4 devices. Delta Chat uses MapLibre and has at least several active Android 4 users in the community and probably many more users who just install and use the application. This change we cannot update MapLibre now: https://github.com/deltachat/deltachat-android/pull/2542

99.2% supported phones mentioned in the changelog is too low, breaking the application for nearly every 100th user is a lot of devices actually, and we likely have more old phones users than an average application.

**Comment by louwers** [comment]
@link2xt Thanks for pitching in! I converted this issue to a discussion. I understand that it is frustrating to drop support for older devices.

We are in the process of using OpenGL ES 3.0. For that we need API level 18. OpenGL ES 3.1 is under consideration, which requires API level 21. See #923. The minimum would really be 18 I think (Android 4.3).

Other than that, I don't know if we can go back to API level 18. Or if there are other good reasons to stick to 21. Have you tried to build it using API level 18?

@artakka Could you share your thoughts?

**Reply by link2xt** [reply]
OpenGL ES 3.0 requirement is even worse as I commented in #923, dropping 6% of devices is not something we may consider. We will rather stay on the old version, look for other options like running [Leaflet](https://leafletjs.com/) in a WebView or drop location streaming support for now if nothing else works.

Dropping old devices may be feasible for a commercial tourist, hiking, hotel booking or similar application, but for a messenger like Delta Chat dropping old devices undermines the purpose of the app, it should keep supporting old devices no matter what, *especially* old devices because this is what people use to connect to their family in remote countries.

**Reply by link2xt** [reply]
> Other than that, I don't know if we can go back to API level 18. Or if there are other good reasons to stick to 21. Have you tried to build it using API level 18?

Haven't tried to compile with override, but in any case level 16 is the minimum we want to support. We have previously tried to bump the requirement from 16 to 18 and had to revert due to multiple bug reports like https://support.delta.chat/t/delta-chat-version-for-android-4-1/940, users running various custom builds etc.

**Reply by louwers** [reply]
> it should keep supporting old devices no matter what, especially old devices because this is what people use to connect to their family in remote countries.

Yes I understand this is a concern for Delta Chat. MapLibre's rendering logic is quite complicated and implementing support for another backend requires some sacrifices in terms of compatibility. *Maybe* it's possible to bring back support for older versions of OpenGL after the renderer modularization effort is complete. But for now I would recommend sticking to an older version!
